pkg/reconciler/v1alpha1/taskrun/resources/input_resource_test.go

```diff
@@ -0,0 +1,313 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	v1alpha1 "github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+	fakeclientset "github.com/knative/build-pipeline/pkg/client/clientset/versioned/fake"
+	informers "github.com/knative/build-pipeline/pkg/client/informers/externalversions"
+	listers "github.com/knative/build-pipeline/pkg/client/listers/pipeline/v1alpha1"
+	"github.com/knative/build-pipeline/pkg/logging"
+	buildv1alpha1 "github.com/knative/build/pkg/apis/build/v1alpha1"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	pipelineResourceLister listers.PipelineResourceLister
+	logger                 *zap.SugaredLogger
+)
+
+func setUp() {
+	logger, _ = logging.NewLogger("", "")
+	fakeClient := fakeclientset.NewSimpleClientset()
+	sharedInfomer := informers.NewSharedInformerFactory(fakeClient, 0)
+	pipelineResourceInformer := sharedInfomer.Pipeline().V1alpha1().PipelineResources()
+	pipelineResourceLister = pipelineResourceInformer.Lister()
+
+	res := &v1alpha1.PipelineResource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "workspace",
+			Namespace: "marshmallow",
+		},
+		Spec: v1alpha1.PipelineResourceSpec{
+			Type: "git",
+			Params: []v1alpha1.Param{
+				v1alpha1.Param{
+					Name:  "Url",
+					Value: "https://github.com/grafeas/kritis",
+				},
+			},
+		},
+	}
+	resWithServiceAccount := &v1alpha1.PipelineResource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "workspace-sa",
+			Namespace: "marshmallow",
+		},
+		Spec: v1alpha1.PipelineResourceSpec{
+			Type: "git",
+			Params: []v1alpha1.Param{
+				v1alpha1.Param{
+					Name:  "Url",
+					Value: "https://github.com/grafeas/kritis",
+				},
+				v1alpha1.Param{
+					Name:  "ServiceAccount",
+					Value: "kritis-service-account",
+				},
+			},
+		},
+	}
+	pipelineResourceInformer.Informer().GetIndexer().Add(res)
+	pipelineResourceInformer.Informer().GetIndexer().Add(resWithServiceAccount)
+}
+
+func TestAddResourceToBuild(t *testing.T) {
+	boolTrue := true
+
+	task := &v1alpha1.Task{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "build-from-repo",
+			Namespace: "marshmallow",
+		},
+		Spec: v1alpha1.TaskSpec{
+			Inputs: &v1alpha1.Inputs{
+				Sources: []v1alpha1.Source{
+					v1alpha1.Source{
+						Name: "workspace",
+						Type: "git",
+					},
+				},
+			},
+		},
+	}
+	taskRun := &v1alpha1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "build-from-repo-run",
+			Namespace: "marshmallow",
+		},
+		Spec: v1alpha1.TaskRunSpec{
+			TaskRef: v1alpha1.TaskRef{
+				Name: "simpleTask",
+			},
+			Inputs: v1alpha1.TaskRunInputs{
+				Resources: []v1alpha1.PipelineResourceVersion{
+					v1alpha1.PipelineResourceVersion{
+						ResourceRef: v1alpha1.PipelineResourceRef{
+							Name: "workspace",
+						},
+						Version: "master",
```

Comment:
can you specify something other than master as default is master

Reply:
there is a test `set revision to default value` that doesn't specify any revision and checks it gets back master, is that what you mean?

Reply:
no, there is not test where the Version of Taskrun is set correctly in GitResource Revision.
Right now, we can't tell if Version is set in loop [L46](https://github.com/knative/build-pipeline/blob/8221983749197b85487840a150d9229a394257c7/pkg/reconciler/v1alpha1/taskrun/resources/input_resources.go#L46) or [L57](https://github.com/knative/build-pipeline/blob/8221983749197b85487840a150d9229a394257c7/pkg/reconciler/v1alpha1/taskrun/resources/input_resources.go#L57)

Reply:
ok, I get what you are saying, will add another test